modules/votes/providers/VoteFormActions/VoteFormActionsContext.tsx

```diff
@@ -0,0 +1,209 @@
+  useContext,
+  useMemo,
+  useState,
+} from 'react'
+import {
+  EligibleDelegatorsData,
+  useEligibleDelegators,
+} from 'modules/delegation/hooks/useEligibleDelegators'
+import { useFormVoteSubmit } from 'modules/votes/ui/VoteForm/useFormVoteSubmit'
+import { useFormVoteInfo } from 'modules/votes/ui/VoteForm/useFormVoteInfo'
+import { formatBalance } from 'modules/blockChain/utils/formatBalance'
+import { useGovernanceSymbol } from 'modules/tokens/hooks/useGovernanceSymbol'
+
+import { ResultTx } from 'modules/blockChain/types'
+import { VoteMode } from 'modules/votes/types'
+
+import invariant from 'tiny-invariant'
+
+import {
+  AttemptCastVoteAsDelegateEventObject,
+  CastVoteEventObject,
+} from 'generated/AragonVotingAbi'
+import { BigNumber } from '@ethersproject/bignumber'
+import {
+  FinishHandler,
+  TransactionSender,
+} from 'modules/blockChain/hooks/useTransactionSender'
+
+export enum VotedAs {
+  delegate = 'delegate',
+  owner = 'owner',
+}
+
+export type VoteFormActionsContextValue = {
+  voteId: string
+  successTx: ResultTx | null
+  setSuccessTx: React.Dispatch<React.SetStateAction<ResultTx | null>>
+  formVoteSubmitData: ReturnType<typeof useFormVoteSubmit>
+  setVoteId: React.Dispatch<React.SetStateAction<string>>
+  ownVotePower: Number
+  handleVote: (mode: VoteMode) => Promise<void>
+  handleDelegatesVote: (
+    mode: VoteMode,
+    selectedAddresses: string[],
+  ) => Promise<void>
+  votedAs: VotedAs | null
+  mode: VoteMode | null
+  txVote: TransactionSender
+  txDelegatesVote: TransactionSender
+  eventsVoted: CastVoteEventObject[] | undefined
+  eventsDelegatesVoted: AttemptCastVoteAsDelegateEventObject[] | undefined
+  eligibleDelegatedVotingPower: BigNumber
+  delegatedVotersAddresses: string[]
+  eligibleDelegatedVoters: EligibleDelegatorsData['eligibleDelegatedVoters']
+  eligibleDelegatedVotersAddresses: string[]
+  delegatorsVotedThemselves: CastVoteEventObject[] | undefined
+  governanceSymbol: string | undefined
+  votedByDelegate: EligibleDelegatorsData['eligibleDelegatedVoters']
+}
+
+const VoteFormActionsContext =
+  createContext<VoteFormActionsContextValue | null>(null)
+
+export const useVoteFormActionsContext = () => {
+  const value = useContext(VoteFormActionsContext)
+  invariant(
+    value,
+    'useVoteFormActionsContext was used outside the VoteFormActionsContext provider',
+  )
+  return value
+}
+
+export const VoteFormActionsProvider: React.FC = ({ children }) => {
+  const [voteId, setVoteId] = useState<string>('')
+  const [successTx, setSuccessTx] = useState<ResultTx | null>(null)
+  const [votedAs, setVotedAs] = useState<VotedAs | null>(null)
+  const [mode, setMode] = useState<VoteMode | null>(null)
+
+  const { data: eligibleDelegatorsData, mutate } = useEligibleDelegators({
+    voteId,
+  }) as { data: EligibleDelegatorsData; mutate: () => Promise<void> }
+
+  const formVoteInfoData = useFormVoteInfo({ voteId })
+  const { data: governanceSymbol } = useGovernanceSymbol()
```

Comment:
There is no need to put governanceSymbol in this context if it is not used here, because the `useGovernanceSymbol` hook is already swr-cached in case of multiple-call. I do understand that it can be handy in some cases to take it from the context and do not write extra hook, but also it bloat up the context with the extra data which is not really necessary for this context internally.

Reply:
fixed in 4604a017354cc8efeeff5ce738df3a4613a574df